### PR TITLE
Benchmarking scripts, load tests and bug fixes

### PR DESCRIPTION
**Pull request by kaustubh-d**
Benchmarking scripts, load tests and bug fixes

### CODE REVIEW

File: s3/replication/common/tests/system/load_test_async_put.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+from s3replicationcommon.s3_site import S3Site
+from s3replicationcommon.s3_session import S3Session
+from s3replicationcommon.s3_put_object import S3AsyncPutObject
+
+
+async def main():
+
+    config = Config()
+
+    bucket_name = "sourcebucket"
```
Comment (rajkumarpatel2602) [inline s3/replication/common/tests/system/load_test_async_put.py:39]: Do we want to have this as a config, like we have done for other load transfer tests???

File: s3/replication/common/tests/system/s3_in_memory.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+    # Setup application routes.
+    app.add_routes(routes)
+
+    # Setup startup/shutdown handlers
+    app.on_startup.append(on_startup)
+    app.on_shutdown.append(on_shutdown)
+
+    # Start the REST server.
+    web.run_app(app, host="0.0.0.0", port="80")
```
Comment (rajkumarpatel2602) [inline s3/replication/common/tests/system/s3_in_memory.py:91]: is it a placeholder for some IP?

File: s3/replication/common/src/s3replicationcommon/s3_session.py [hunk 3]
```diff
@@ -31,7 +32,8 @@ def __init__(self, logger, s3_site, access_key, secret_key):
         self.access_key = access_key
         self.secret_key = secret_key
 
-        self._client_session = aiohttp.ClientSession()
+        connector = aiohttp.TCPConnector(limit=number_of_connections)
```
Comment (rajkumarpatel2602) [inline s3/replication/common/src/s3replicationcommon/s3_session.py:35]: Do we require similar changes at other 'aiohttp.ClientSession' instances as well?

File: s3/replication/common/tests/system/s3_in_memory.py (hunk 2 above)
Comment (kaustubh-d) [reply s3/replication/common/tests/system/s3_in_memory.py:91]: This is so we can listen on all IPs and its ok in test environment, we will not be using this in prod.

File: s3/replication/common/src/s3replicationcommon/s3_session.py (hunk 3 above)
Comment (kaustubh-d) [reply s3/replication/common/src/s3replicationcommon/s3_session.py:35]: This is currently for all S3 connections, for other paths (manager <--> replicator) we will add in future and will need to be tested separately.

File: s3/replication/common/tests/system/load_test_async_put.py (hunk 1 above)
Comment (kaustubh-d) [reply s3/replication/common/tests/system/load_test_async_put.py]: Fixed.

**Review by rajkumarpatel2602 — APPROVED** [review]

**Pull request merged**